modules/calendar/includes/get_provider_events.php

```diff
@@ -24,7 +24,44 @@
 
 $fetchedEvents = fetchAllEvents($_POST['start'], $_POST['end']);
 
+// facility ACL check
+if ($GLOBALS['facility_acl'] == 1) {
+  $facility_filter = do_action('filter_patient_select_pnuserapi', $_SESSION['authUser']);
+  // do_action returns either
+  // 1. $where = " pd.facility IN ( $facilityString ) "; i.e. schedule/allowed facility ids in a SQL string or
+  // 2. $where = " pd.facility = '-1' "; when logged in user has no corresponding facilities in users_facility table i.e. no allowed facilities
+
+  $facilities_allowed_to_user = array();  // array to contain facility ids which logged in user is allowed to see
```

Comment:
@aethelwulffe I think the reason for not displaying any appointment is that in all cases, they are getting `continue;`d. Please move this array definition to somewhere before line 27 so that it's available in both if blocks and check again.

Reply:
Moved line.  Debug out SELECT facility.id AS fid\r\n            FROM facility\r\n            WHERE facility.id IN ( 4,4 )
actual settings:
![image](https://user-images.githubusercontent.com/553751/42065884-9cb6dc3c-7b0b-11e8-8946-5b0896f208ce.png)

No display of appointments.

Reply:
Multi-select for schedule facilities not working with the feature off.  Does not properly echo changes when on.  Adds additional facilities when there are a total of 3 (adding 2)

Reply:
In the user code, right. May be there needs to be another chapter to this novel.

Reply:
Multi-select for schedule facilities has another issue I think, when there you're logged in as a user and there is no row in users_facility table, the multi-select still shows user's default facility as **greyed** or selected but there's no mention in that table.

It's because of `<option <?php echo in_array($frow['id'], $ufid) || $frow['id'] == $iter['facility_id'] ? "selected" : null ?> value="<?php echo $frow['id'] ?>"><?php echo htmlspecialchars($frow['name']) ?></option>` where $iter['facility_id'] is id of default facility of selected user.